propagation/http_trace_context_propagator.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package propagation
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"net/textproto"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/api/core"
+	apipropagation "go.opentelemetry.io/api/propagation"
+	"go.opentelemetry.io/api/tag"
+)
+
+const (
+	supportedVersion  = 0
+	maxVersion        = 254
+	traceparentHeader = "traceparent"
+)
+
+type httpTraceContextPropagator struct{}
+
+var _ apipropagation.TextFormatPropagator = httpTraceContextPropagator{}
+
+// CarrierExtractor implements CarrierExtractor method of TextFormatPropagator interface.
+//
+// It creates CarrierExtractor object and binds carrier to the object. The carrier
+// is expected to be *http.Request. If the carrier type is not *http.Request
+// then an empty extractor. Extract method on empty extractor does nothing.
+func (hp httpTraceContextPropagator) CarrierExtractor(carrier interface{}) apipropagation.Extractor {
+	req, ok := carrier.(*http.Request)
+	if ok {
+		return traceContextExtractor{req: req}
+	}
+	return traceContextExtractor{}
```

Comment:
Maybe the propagation package, or API propagation package could provide a noop implementations of the interfaces? Then you could use it here and assume that `traceContextExtractor`'s `req` field can't be `nil`?